lib/ex_doc/formatter/epub.ex

```diff
@@ -34,20 +30,25 @@ defmodule ExDoc.Formatter.EPUB do
     datetime = format_datetime()
     nodes = modules ++ exceptions ++ protocols
 
-    generate_content(output, config, nodes, uuid, datetime)
-    generate_toc(output, config, nodes, uuid)
-    generate_nav(output, config, nodes)
-    generate_title(output, config)
-    generate_list(output, config, modules)
-    generate_list(output, config, exceptions)
-    generate_list(output, config, protocols)
+    generate_content(config, nodes, uuid, datetime)
+    generate_toc(config, nodes, uuid)
+    generate_nav(config, nodes)
+    generate_title(config)
+    generate_list(config, modules)
+    generate_list(config, exceptions)
+    generate_list(config, protocols)
 
-    {:ok, epub_file} = generate_epub(output)
-    File.rm_rf!(output)
+    {:ok, epub_file} = generate_epub(config.output)
+    File.rm_rf!(config.output)
     epub_file
   end
 
   defp normalize_config(config) do
+    output =
+      config.output
+      |> Path.expand()
+      |> Path.join("#{config.project}-v#{config.version}")
+    config = Map.put(config, :output, output)
```

Comment:
`%{config | output: output}` would be semantically better (as the key `:output` is guaranteed to exist and is being updated).